app/core/Tracker2/BaseTracker.js

```diff
@@ -0,0 +1,108 @@
+import EventEmitter from 'events'
+
+/**
+ * A baseline tracker that:
+ *   1. Defines a standard initialization flow for all trackers
+ *   2. Exposes an event emitter interface for all trackers
+ *   3. Exposes a default tracker interface with a noop implementation.
+ *
+ * Standard initialization flow:
+ * The standard initialization flow is based around the this.initializationComplete promise
+ * that is exposed publicly by the tracker.  This promise should be resolved by the tracker
+ * by calling this.onInitializationSuccess (or this.onInitializationError) when the
+ * initialization process has been completed (or failed).  The initializationComplete
+ * promise should _never_ be overwritten as other external components can depend on it.
+ */
+export default class BaseTracker extends EventEmitter {
+  constructor () {
+    super()
+
+    this.initializing = false
+    this.initialized = false
+
+    this.setupInitialization()
+  }
+
+  async identify (traits = {}) {}
+
+  async trackPageView (includeIntegrations = {}) {}
+
+  async trackEvent (action, properties = {}, includeIntegrations = {}) {}
+
+  async trackTiming (duration, category, variable, label) {}
+
+  get isInitialized () {
+    return this.initialized
+  }
+
+  set isInitialized (initialized) {
+    return this.initialized = initialized
+  }
+
+  get isInitializing () {
+    return this.initializing
+  }
+
+  set isInitializing (initializing) {
+    return this.initializing = initializing
+  }
+
+  get initializationComplete () {
+    return this.initializationCompletePromise
+  }
+
+  /**
+   * Standard initialization method to be used to initialize the tracker.  Ensures that the
+   * initialize process is only called once and returns the initializationComplete promise.
+   *
+   * Calls internal initialize method
+   */
+  async initialize () {
+    if (this.isInitializing) {
+      return this.initializationComplete
+    }
+
+    this.isInitializing = true
+
+    await this._initializeTracker()
```

Comment:
Unsure if this is awaiting anything as `_initializeTracker` doesn't appear to return the `onInitializeSuccess` promise.

Reply:
See https://github.com/codecombat/codecombat/pull/5776#discussion_r391345977 for more detail.  `onInitializeSuccess` is just a callback function that resolves the top level init promise.  Maybe there is a better name we can us?  `notifyTrackerInitialized`?  I struggled with a good name for this.